WebSocket 2.2.0 release

markt-asf opened the pull request:
## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [X] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of jakarta-wombat-spec-x.y.pdf
- [x] Specification HTML in the form of jakarta-wombat-spec-x.y.html
- [X] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [X] For a [Release Review](https://www.eclipse.org/projects/handbook/#release-review), a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification. The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
   - [x] Updated release record
   -  ~[ ] Generated IP Log~ *no longer required according to project handbook*
   - [x] Email to PMC
   - [x] Start release review by emailing EMO 
- [X] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/repositories/jakartawebsocket-1026/
- [X] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      https://www.eclipse.org/downloads/download.php?file=/ee4j/jakartaee-tck/jakartaee11/staged/eftl/jakarta-websocket-tck-2.2.0.zip
- [X] The URL of the compatibility certification request issue:
      Tomcat: https://github.com/jakartaee/websocket/issues/427
      Tyrus: https://github.com/jakartaee/websocket/issues/425 and https://github.com/jakartaee/websocket/issues/426 
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

Note: If any item does not apply, check it and mark N/A below it.

markt-asf commented:
I'm expecting to fill in the blanks over the next 24 hours or so.

kazumura commented:
# Spec Review Checklist

1. Spec PR
  - [ ] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [ ] Directory of form {spec}/x.y
  - [ ] PDF of form jakarta-{spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [ ] HTML of form jakarta-{spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [ ] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta.{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to project release plan of the form https://projects.eclipse.org/projects/ee4j.{spec}/releases/x.y/plan
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip (The folder path is required, the file name pattern is preferred.)
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation
  - [ ] Review summary statement to ensure it has been updated to reflect the work accomplished and does not contain proposal statements as might be found in Plan Review statement.

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request

10. If a Release Review is required, the specification project team contacts the EMO to initiate the review by sending an email to emo@eclipse.org.
    (A Release Review is not required if the current release is a Service Release based on a previously successful Major or Minor
    release as indicated by a release record on the project's Releases page, e.g., the [Jakarta Servlet releases page](https://projects.eclipse.org/projects/ee4j.servlet/reviews).)
    - [ ] An issue will be created by the EMO to track the release review.
    - [ ] The specification project team requests approval for the release from the PMC by sending an email to ee4j-pmc@eclipse.org.
    - [ ] The specification project team then delivers an IP Log to the IP Team for their review as described in https://www.eclipse.org/projects/handbook/#pmi-commands-iplog.

11. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

kazumura reviewed (changes requested):
@markt-asf  Thanks fro PR.

* The name of specification documentation must be
"jakarta-websocket-spec-2.2.{html|pdf}" but not "websocket-spec-2.2.{html|pdf}".

* Do not include META-INF directories in the PR.

* Copyright years in API doc footer must be "2018, 2024", but not "2018, 2022".

* Use v1.1 of EFSL but not v1.0. (in specification documents and api doc)

* Do not modify the text of EFSL. (you do not need to change place holders)
For reference see "https://www.eclipse.org/lists/jakarta.ee-wg/msg00996.html".

* Include  v1.1 of Technology Compatibility Kit License v but not v1.0. (LICENSE.md in TCK zip file)

* In README.txt in TCK zip file,
"2. Download and install the WebSocket 2.1 Compatible Implementation/jarfile." must be "... WebSocket 2.2 ...".

markt-asf commented:
I've fixed the non-TCK issues, restaged the 2.2.0 release and updated this PR.
I have created https://github.com/jakartaee/platform-tck/pull/1271 for the TCK issues although I'm not sure how to address the TCK license issue.

markt-asf commented:
All the required material should now be available.
Note that the certification request is for a 11.0.0-M19-SNAPSHOT release of Tomcat. I expect to update this to the 11.0.0-M19 release once that is available early/mid in April.

kazumura commented:
@markt-asf Thanks for updating.

* Please also replace the contents of "websocket/2.2/apidocs/{client|server}/doc-files/EFSL.html" with v1.1.
* The staging repo is changed to https://jakarta.oss.sonatype.org/content/repositories/jakartawebsocket-1025. Please change the PR template.
* Please add "certification" label to CCR issue.

markt-asf commented:
EFSL for Javadoc updated and WebSocket 2.2 restaged to pick up that change.
Javadoc copy and staging repo link updated.
Label added to certification request

markt-asf commented:
Note: The TCK being used still has SNAPSHOT dependencies. This review should not progress until an updated TCK is available without SNAPSHOT dependencies.

Pandrex247 commented on websocket/2.2/_index.md:
```diff
@@ -9,35 +9,34 @@ This release removes references to the SecurityManager and provides some minor u
+
+* Add the `getSession()` method to `SendResult`.
+
+* Clarify the behaviour if `@OnMessage.maxMessageSize` it is set to a value larger than `Integer.MAX_VALUE`.
 
 ### Removals, deprecations or backwards incompatible changes
-* All references to the SecurityManager have been removed
+* None
```
Should line 8 also be adjusted if the security manager hasn't been removed?

> This release removes references to the SecurityManager and provides some minor updates and clarifications.

markt-asf replied:
Thanks for spotting the inconsistency. This is an odd one. If you look at the code history, there were no references removed. However, the 2.1.1 release added a reference to the SecurityManager on the 2.1.x branch. From a user's perspective it does look like that reference has been removed so I'll restore the previous version of line 19 and keep line 8 as is.

Pandrex247 commented on websocket/2.2/_index.md:
```diff
@@ -9,35 +9,34 @@ This release removes references to the SecurityManager and provides some minor u
 # Compatible Implementations
 
-* Eclipse Tyrus (TBC)
+* Apache Tomcat 11.0.0-M19-SNAPSHOT
```
This should be a download link, not just a name.
The snapshot link on the CCR no longer seems to be valid

kazumura replied:
@markt-asf   
Thanks for updating PR.

* The URL of Tyrus is TCK result, but not Tyrus download link. Please change it to the download link.
* Please also update CCR links of this PR template to the new one https://github.com/jakartaee/websocket/issues/427,
and add Tyrus CCR.

kazumura replied:
@markt-asf   
Thanks for updating Tyrus links.

The CCR links you added are not valid for this PR, because the TCK SHA values are different form TCK binary in this PR.  
So, Tyrus CCR are needed to update as well as Tomcat M20.

markt-asf replied:
The PR uses the SHA512 since that is the only hash that the download page presents. The SHA-512 is:
5bb92094db8d46f60f561e0bcb81146b4802072abc39ce059bb69a63b310d3e2c970ebf8b1a5fbcc97f8410f4d05918ac2b21c847d89af027e9191d0e7e45a20

This is the same as the SHA-512 listed on the Tomcat CCR.

The SHA-256 is (obtained by downloading the file, confirming the SHA-512 hash and the calculating the SHA-256 hash):
ff50825305bd1318a4e6bdfab21b829cfd8cf89fab8a970f007b52636d3f2629

The Tyrus CCR references the correct SHA-256 hash.

I am not seeing any inconsistencies here apart from the download page presenting the SHA-512 hash but the PR checklist expecting the SHA-256 hash. The WebSocket project has approved the Tyrus CCR having confirmed the SHA-256 hash.

I do not see any changes that are required for this to proceed.

kazumura replied:
@markt-asf   
I didn't notice the difference SHA-256 and SHA-512. Thanks for clarification.

mtdelgadoa commented:
Hello all, 
the EMO is trying to avoid duplication of release records, so we will be providing our feedback and comments related to this release directly here.

EDP Review status: ONGOING

### **EMO review checklist**
PMI record: https://projects.eclipse.org/projects/ee4j.websocket/reviews/2.2.0-release-review

_**EF Specification Process**_      
- [ ] Spec Committee Ballot completed

_**Intellectual Property Management**_      
- [ ] All project code has [copyright and license headers](https://www.eclipse.org/projects/handbook/#ip-copyright-headers) correctly applied. ** EMO will scan the code at their discretion **   
- [ ]  All distributed third-party content has been vetted by the IP Due Diligence process (i.e., IP Log has been approved)   

_**Open Source Rules of Engagement**_
- [ ] PMC approval [on the mailing list](https://github.com/eclipsefdn/emo-team/emo/-/blob/main/%7Blink%20to%20the%20message%7D)

General:
- [ ] Project is operating within the mission and scope defined in its top-level project’s charter   
- [ ]  Project is operating within the bounds of its own scope   
- [ ]  Project is operating in an open and transparent manner       
- [ ]  Overall the project is operating according to the Eclipse Development Process.   

Things to check:
- Communication channels advertised
- Advertised communication channels used
- Committers are responding to questions
- Committers are responding to issues
- Committers are responding to pull/merge/review requests

_**Branding and Trademarks**_
The following applies when the project has a custom website.
To the best of our knowledge:
- [ ]  Project content correctly uses Eclipse Foundation trademarks    
- [ ] Project content (code and documentation) does not violate trademarks owned by other organizations   

Things to check:
- [ ]  Project website uses the [project's formal name](https://www.eclipse.org/projects/handbook/#trademarks-website-name) in first and all prominent references
- [ ]   Project website includes a trademark attribution statement
- [ ] Project website footers contain all [necessary elements](https://www.eclipse.org/projects/handbook/#trademarks-website-footer)

_**Legal Documentation**_
Required files:   
- [ ] License files in all repository roots   
- [ ]  README       
- [ ]  CONTRIBUTING (or equivalent)   

Recommended files:
- [ ] NOTICES or equivalent (you can use the [Legal Documentation generator](https://www.eclipse.org/projects/tools/documentation.php) available under committer tools)   
- [ ]  CODE_OF_CONDUCT   
- [ ]  SECURITY   
 
See examples for [Security file](https://gitlab.eclipse.org/security/best-practices/-/blob/main/templates/SECURITY.md) and [Code of Conduct](https://github.com/eclipse/.github/blob/master/CODE_OF_CONDUCT.md).

Required elements:
- [ ]  ECA is referenced/described   

Recommended elements:
- [ ]  Build instructions provided   
- [ ] [Security policy](https://www.eclipse.org/security/policy.php) is described   
 
_**Metadata (PMI)**_
- [ ] The formal name, e.g. "Eclipse Foo™", is used in the project title   
- [ ] The formal name including appropriate marks (e.g, "™") is used in the first mention in the text of the project description, and scope   
- [ ] The project description starts with a single paragraph that can serve as an executive summary   
- [ ] Source code repository references are up-to-date   
- [ ] Download links and information are up-to-date (see [EF handbook for more information on how-to do this](https://www.eclipse.org/projects/handbook/#resources-downloads))   
- [ ] Communication channels listed in the PMI (i.e. public mailing list, forums, etc.)

kazumura commented:
@markt-asf   

> Note: The TCK being used still has SNAPSHOT dependencies. This review should not progress until an updated TCK is available without SNAPSHOT dependencies.

I withdraw the current ballot.
I will restart the ballot after TCK and CCR will be updated.

markt-asf commented:
I believe the issues have all been resolved and the release ballot can be started.

kazumura commented:
ballot completed.
https://www.eclipse.org/lists/jakarta.ee-spec/msg03387.html
Thanks @markt-asf

kazumura commented:
On ballot completion, the specification committee mentor:

 - [x] adds this final checklist to the main PR.
 - [x] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [x] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [ ] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [x] creates an issue in the specification project that includes the following checklist for the specification project team:
https://github.com/jakartaee/websocket/issues/428
